frontend/src/App.tsx

```diff
@@ -9,8 +9,15 @@ import SignUp from '@/pages/signup-page';
 import AdminUsers from '@/pages/admin-users';
 import AdminBlogs from '@/pages/admin-blogs';
 import AdminContainer from './components/admin-container';
+import { useEffect } from 'react';
 
 function App() {
+
+  useEffect(() => {
+    const storedTheme = localStorage.getItem('theme');
```

Comment:
useLayoutEffect may be better? Kindly see which would be more relevant. And can you even evaluate and optimize the toggle-button file, if there is scope? I feel it has some uneccessary code. 

And it is better to have this logic somewhere else and import in the main file than here, helps in seperation of concerns. Please do some research and get back, happy to know your inputs

Reply:
Ok Sir